File: app/src/main/java/nl/viasalix/btroosterlite/MainActivity.java
/*  BTRooster Lite: Roosterapp voor Calvijn College
 *  Copyright (C) 2017 Rutger Broekhoff <rutger broekhoff three at gmail dot com>
 *                 and Jochem Broekhoff
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package nl.viasalix.btroosterlite;

import android.app.FragmentTransaction;
import android.graphics.Color;
import android.app.Fragment;
import android.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.aurelhubert.ahbottomnavigation.AHBottomNavigation;
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationItem;

public class MainActivity extends AppCompatActivity {
    public static final String AUTHORITY = "1-dot-0-dot-btrfrontend.appspot.com";
    Fragment currentFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        AHBottomNavigation bottomNavigation = findViewById(R.id.bottom_navigation);

        AHBottomNavigationItem item_timetable = new AHBottomNavigationItem("Rooster", R.drawable.ic_border_all_black_24dp, R.color.colorBottomNavigationPrimary);
        AHBottomNavigationItem item_cup = new AHBottomNavigationItem("CUP", R.drawable.ic_event_black_24dp, R.color.colorBottomNavigationPrimary);
        AHBottomNavigationItem item_test_timetable = new AHBottomNavigationItem("Toetsrooster", R.drawable.ic_chrome_reader_mode_black_24dp, R.color.colorBottomNavigationPrimary);
        bottomNavigation.addItem(item_timetable);
        bottomNavigation.addItem(item_cup);
        bottomNavigation.addItem(item_test_timetable);
        bottomNavigation.setDefaultBackgroundColor(Color.parseColor("#FEFEFE"));
        bottomNavigation.setBehaviorTranslationEnabled(false);
        bottomNavigation.setAccentColor(Color.parseColor("#F63D2B"));
        bottomNavigation.setInactiveColor(Color.parseColor("#747474"));
        bottomNavigation.setForceTint(true);
        bottomNavigation.setTitleState(AHBottomNavigation.TitleState.ALWAYS_SHOW);
        bottomNavigation.setCurrentItem(0);

        tabSelected(bottomNavigation.getCurrentItem(), false);

        if (currentFragment instanceof TimetableFragment) {
            bottomNavigation.setCurrentItem(0);
        } else if (currentFragment instanceof CUPFragment) {
            bottomNavigation.setCurrentItem(1);
        } else if (currentFragment instanceof TestTimetableFragment) {
            bottomNavigation.setCurrentItem(2);
        }

        bottomNavigation.setOnTabSelectedListener((position, wasSelected) -> {
            tabSelected(position, wasSelected);

            return true;
        });
    }

    private void tabSelected(int position, boolean wasSelected) {
        if (!wasSelected) {
            switch (position) {
                case 0:
                    launchTimetableFragment();
                    break;
                case 1:
                    launchCUPFragment();
                    break;
                case 2:
                    launchTestTimetableFragment();
                    break;
                default:
                    launchTimetableFragment();
            }
        }
    }


    private void launchTimetableFragment() {
        currentFragment = null;
        currentFragment = new TimetableFragment();
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragment_container, currentFragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }

    private void launchCUPFragment() {
        currentFragment = null;
        currentFragment = new CUPFragment();
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragment_container, currentFragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }

    private void launchTestTimetableFragment() {
        currentFragment = null;
        currentFragment = new TestTimetableFragment();
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragment_container, currentFragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/nl/viasalix/btroosterlite/activities/MainActivity.kt
/*  BTRooster Lite: Roosterapp voor Calvijn College
 *  Copyright (C) 2017 Rutger Broekhoff <rutger broekhoff three at gmail dot com>
 *                 and Jochem Broekhoff
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package nl.viasalix.btroosterlite.activities

import android.app.Fragment
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import com.aurelhubert.ahbottomnavigation.AHBottomNavigation
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationAdapter
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationItem
import nl.viasalix.btroosterlite.R
import nl.viasalix.btroosterlite.cup.CUPIntegration
import nl.viasalix.btroosterlite.fragments.CUPFragment
import nl.viasalix.btroosterlite.fragments.TestTimetableFragment
import nl.viasalix.btroosterlite.fragments.TimetableFragment
import nl.viasalix.btroosterlite.introduction.IntroductionActivity
import nl.viasalix.btroosterlite.singleton.Singleton
import org.jetbrains.anko.design.snackbar

class MainActivity : AppCompatActivity() {
    private var currentFragment: Fragment? = null
    private var bottomNavigation: AHBottomNavigation? = null

    private var backPressed: Long = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bottomNavigation = findViewById(R.id.bottom_navigation)

        val tabColors: IntArray = intArrayOf(
                ContextCompat.getColor(this, R.color.colorBottomNavigationPrimary),
                ContextCompat.getColor(this, R.color.colorBottomNavigationPrimary),
                ContextCompat.getColor(this, R.color.colorBottomNavigationPrimary))

        val navigationAdapter = AHBottomNavigationAdapter(this, R.menu.bottom_navigation)
        navigationAdapter.setupWithBottomNavigation(bottomNavigation, tabColors)

        bottomNavigation!!.defaultBackgroundColor = Color.parseColor("#FEFEFE")
        bottomNavigation!!.isBehaviorTranslationEnabled = false
        bottomNavigation!!.accentColor = Color.parseColor("#F63D2B")
        bottomNavigation!!.inactiveColor = Color.parseColor("#747474")
        bottomNavigation!!.isForceTint = true
        bottomNavigation!!.titleState = AHBottomNavigation.TitleState.ALWAYS_SHOW

        currentFragment = null

        if (PreferenceManager.getDefaultSharedPreferences(this)
                        .getBoolean("firstLaunch", true)) {
            startActivity(Intent(this, IntroductionActivity::class.java))
            finish()
        }

        bottomNavigation!!.setOnTabSelectedListener { position, wasSelected ->
            tabSelected(position, wasSelected)
            true
        }

        Singleton.cupIntegration = CUPIntegration(this)
    }

    override fun onResume() {
        super.onResume()

        tabSelected(bottomNavigation!!.currentItem, false)
    }

    override fun onBackPressed() {
        // Interval is 2000 seconden
        if (backPressed + 2000 > System.currentTimeMillis()) {
            super.onBackPressed()
            return
        } else {
            snackbar(findViewById(android.R.id.content), resources.getString(R.string.tap_again_exit))
        }

        backPressed = System.currentTimeMillis()
    }

    private fun tabSelected(position: Int, wasSelected: Boolean) {
        if (!wasSelected) {
            when (position) {
                0 -> launchTimetableFragment()
                1 -> launchCUPFragment()
                2 -> launchTestTimetableFragment()
                else -> launchTimetableFragment()
            }
        }
    }

    fun launchTimetableFragment() {
        currentFragment = null
        currentFragment = TimetableFragment()

        val fragmentManager = fragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()

        fragmentTransaction.replace(R.id.fragment_container, currentFragment)
        fragmentTransaction.commit()

        bottomNavigation!!.setCurrentItem(0, false)
    }

    private fun launchCUPFragment() {
        currentFragment = null
        currentFragment = CUPFragment()

        val fragmentManager = fragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()

        fragmentTransaction.replace(R.id.fragment_container, currentFragment)
        fragmentTransaction.commit()

        bottomNavigation!!.setCurrentItem(1, false)
    }

    private fun launchTestTimetableFragment() {
        currentFragment = null
        currentFragment = TestTimetableFragment()

        val fragmentManager = fragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()

        fragmentTransaction.replace(R.id.fragment_container, currentFragment)
        fragmentTransaction.commit()

        bottomNavigation!!.setCurrentItem(2, false)
    }

    companion object {
        const val AUTHORITY = "1d2d0-dot-btrfrontend.appspot.com"
        const val SCHEME = "https"

        // In-house testing
//        const val SCHEME = "http"
//        const val AUTHORITY = "192.168.178.73:8080"
    }
}
